react/__tests__/DeviceCoordinates.test.tsx

```diff
@@ -0,0 +1,164 @@
+import React from 'react'
+import { fireEvent, render, screen, waitFor } from '@vtex/test-tools/react'
+import { MockedProvider, MockedResponse } from '@apollo/react-testing'
+import { Address } from 'vtex.checkout-graphql'
+import { AddressRules } from 'vtex.address-context/types'
+
+import { AddressContextProvider } from '../__mocks__/vtex.address-context/AddressContext'
+import DeviceCoordinates from '../DeviceCoordinates'
+import { reverseGeocode } from '../__fixtures__/graphql/reverseGeocode.fixture'
+import { vtexOfficeAddress } from '../__fixtures__/address.fixture'
+
+interface RenderComponentProps {
+  deviceCoordinatesProps?: React.ComponentProps<typeof DeviceCoordinates>
+  address?: Address | null
+  countries?: string[]
+  rules?: AddressRules
+  graphqlMocks?: MockedResponse[]
+}
+
+const defaultGraphqlMocks = [] as MockedResponse[]
+
+describe('Device Coordinates', () => {
+  const renderComponent = ({
+    deviceCoordinatesProps,
+    address = { country: 'BRA' },
+    countries = [],
+    rules = {},
+    graphqlMocks = [],
+  }: RenderComponentProps) => {
+    return render(
+      <MockedProvider
+        addTypename={false}
+        mocks={[...defaultGraphqlMocks, ...graphqlMocks]}
+      >
+        <AddressContextProvider
+          address={address}
+          countries={countries}
+          rules={rules}
+        >
+          <DeviceCoordinates {...deviceCoordinatesProps} />
+        </AddressContextProvider>
+      </MockedProvider>
+    )
+  }
```

Comment:
Although this is very similar to LocationSearch renderComponent, I think there is no need to merge this in an util class, at least for now.